db4-storage/src/wal/mod.rs

```diff
@@ -42,5 +40,116 @@ pub trait WalOps {
     /// `cutoff_lsn` acts as a hint for which records can be safely discarded during rotation.
     fn rotate(&self, cutoff_lsn: LSN) -> Result<(), DBV4Error>;
 
+    /// Returns an iterator over the wal entries in the given directory.
     fn replay(dir: impl AsRef<Path>) -> impl Iterator<Item = Result<WalRecord, DBV4Error>>;
 }
+
+// Raphtory-specific logging & replay methods.
+pub trait GraphWal {
+    type Entry: 'static;
+
+    fn log_begin_txn(&self, txn_id: TransactionID) -> Result<LSN, DBV4Error>;
+
+    fn log_end_txn(&self, txn_id: TransactionID) -> Result<LSN, DBV4Error>;
+
+    fn log_add_edge(
+        &self,
+        txn_id: TransactionID,
+        t: TimeIndexEntry,
+        src: VID,
+        dst: VID,
+        layer_id: usize,
+        t_props: &[(usize, Prop)],
+        c_props: &[(usize, Prop)],
+    ) -> Result<LSN, DBV4Error>;
+
+    fn log_node_id(&self, txn_id: TransactionID, gid: GID, vid: VID) -> Result<LSN, DBV4Error>;
```

Comment:
Can we rename the error to StorageError or some other global error that is currently used in pomtetry storage

Reply:
Created an issue: https://github.com/Pometry/pometry-storage/issues/49?issue=Pometry%7Cpometry-storage%7C75